### DISCUSSION TITLE
Different views on the same data objects

### DISCUSSION THREAD

**Discussion by lpheller** [pimcore/pimcore#9475] [Q&A / Support]
Hi Pimcore community,
I'm looking for a way to filter / view data objects based on some provided tags or something similar. 

Basically I need some kind of way to create different views on the same data objects based on some settings. 
Let's imagine there are a bunch of `Products` that are stored and managed in Pimcore. What I'm looking for is a way to split the products up in different views / pre-filtered lists, e.g. by client or project etc.

What I was thinking of is to use  tags like "catalogue2021" or "clientXY" and add them to the products.  
Next I tried to create a custom view named "Catalogue2021" but i can't find anything about how to filter by tags in the custom view definition. 

Now I'm not sure if this even is the best approach or if there are better ways to achieve this goal with Pimcore.

**Comment by NiklasBr** [comment]
Like [Drupal's Views module](https://www.drupal.org/docs/8/core/modules/views)?

**Reply by lpheller** [reply]
Maybe, but I'm not familiar with Drupal so i don't actually know how this would look like. But the description sounds like it may be a fair comparison. 

I think, the custom views shipped with Pimcore already may be a solution, if I'd find a way to include the tag filter within the definition.

**Comment by fashxp** [comment]
in the custom view definition you can create your own select statements and joins and thus also could filter for certain tags. 
see also [here](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Class_Settings/Custom_Views.html)
